### DISCUSSION TITLE
Plans to add unique identifiers to buttons and inputs?

### DISCUSSION THREAD

**Discussion by hananurrehman** [suriya-mca/pimify#61] [Ideas]
Hi, I find your project to be very interesting. I'm currently writing Playwright tests on this and was wondering if you plan to add unique identifiers to certain buttons and input fields in the project?
If not, maybe you can provide me some guidance and I can contribute in that manner?

**Comment by suriya-mca** [comment]
Thanks for showing interest in Pimify! 😊 I appreciate your suggestion about adding unique identifiers for testing. Since Pimify uses Django Unfold for the custom admin panel and components, adding things like id or data-test-id is not easy without changing the default templates.

I’m not very familiar with Playwright. I recommend checking out the Django Unfold documentation to see how it works: https://github.com/unfoldadmin/django-unfold

If you have any ideas or suggestions, feel free to share! I’d love to discuss and work on it together.

**Reply by hananurrehman** [reply]
Hi thanks for the reply.
Just a small question, currently I'm facing an issue where superuser creation doesn't work unless done manually. 
Is it possible to automate superuser creation? 
Note: I'm running the app in docker but my fork may be a bit outdated too.

**Reply by suriya-mca** [reply]
Thanks for letting me know! I’ve fixed the issue. Superuser creation works fine when running scripts manually, but it wasn’t working in Docker, and I couldn’t find a way to automate it.

So, I’ve set up a default superuser:

Username: admin
Password: admin123
You can log in with these and change the password after logging in. I’ve also updated the script and Dockerfile to make this work.

Let me know if you run into any other issues!

**Reply by suriya-mca** [reply]
Hi @hananurrehman, just following up to check if my previous response resolved your issue—let me know!

**Reply by hananurrehman** [reply]
Hi @suriya-mca , sorry for not replying. I couldn't get the chance to look at the project but will hopefully find the time this week.

**Reply by hananurrehman** [reply]
Hi @suriya-mca 
Thanks for the fix! it works.
For now since I haven't explored the exchange rates thing, i had to assign a dummy value in .env
`OPEN_EXCHANGE_RATES_APP_ID=your-app-id-here`
But it works for me now

**Reply by suriya-mca** [reply]
Hi @hananurrehman 

I have an update. I’ve rolled back to the old approach due to some security risks. I’ve removed the default superuser creation. Superuser creation works fine with scripts, but you’ll need to create it manually in Docker as there’s no reliable or secure way to automate it.

Please make sure to update your fork with the latest changes and take a look at the updated README for more details.

Let me know if you have any questions!

**Reply by hananurrehman** [reply]
Sure I will check it out.
